netobserv/v1alpha1/flowcollector_types.go

```diff
@@ -0,0 +1,436 @@
+	// will fetch all the interfaces in the system, excepting the ones listed in ExcludeInterfaces.
+	// If an entry is enclosed by slashes (e.g. `/br-/`), it will match as regular expression,
+	// otherwise it will be matched as a case-sensitive string.
+	//+optional
+	Interfaces []string `json:"interfaces,omitempty"`
+
+	// ExcludeInterfaces contains the interface names that will be excluded from flow tracing.
+	// If an entry is enclosed by slashes (e.g. `/br-/`), it will match as regular expression,
+	// otherwise it will be matched as a case-sensitive string.
+	//+kubebuilder:default=lo;
+	ExcludeInterfaces []string `json:"excludeInterfaces,omitempty"`
+
+	//+kubebuilder:validation:Enum=trace;debug;info;warn;error;fatal;panic
+	//+kubebuilder:default:=info
+	// LogLevel defines the log level for the NetObserv eBPF Agent
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// Env allows passing custom environment variables to the NetObserv Agent. Useful for passing
+	// some very concrete performance-tuning options (e.g. GOGC, GOMAXPROCS) that shouldn't be
+	// publicly exposed as part of the FlowCollector descriptor, as they are only useful
+	// in edge debug/support scenarios.
+	//+optional
+	Env map[string]string `json:"env,omitempty"`
+
+	// Privileged mode for the eBPF Agent container. If false, the operator will add the following
+	// capabilities to the container, to enable its correct operation:
+	// BPF, PERFMON, NET_ADMIN, SYS_RESOURCE.
+	// +optional
+	Privileged bool `json:"privileged,omitempty"`
+}
+
+// FlowCollectorKafka defines the desired Kafka config of FlowCollector
+type FlowCollectorKafka struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//+kubebuilder:default:=false
+	// Set true to use Kafka as part of the flow collection pipeline. When enabled, the pipeline is split in two parts: ingestion and transformation, connected by Kafka.
+	// The ingestion is either done by a specific flowlogs-pipeline workload, or by the eBPF agent, depending on the value of `spec.agent`.
+	// The transformation is done by a new flowlogs-pipeline deployment.
+	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=""
+	// Address of the Kafka server
+	Address string `json:"address"`
+
+	//+kubebuilder:default:=""
+	// Kafka topic to use. It must exist, NetObserv will not create it.
+	Topic string `json:"topic"`
+}
+
+// FlowCollectorFLP defines the desired flowlogs-pipeline state of FlowCollector
+type FlowCollectorFLP struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//+kubebuilder:validation:Enum=DaemonSet;Deployment
+	//+kubebuilder:default:=DaemonSet
+	// Kind is the workload kind, either DaemonSet or Deployment
+	// When using Kafka, this option only affects the flowlogs-pipeline ingester, not the transformer.
+	Kind string `json:"kind,omitempty"`
+
+	//+kubebuilder:validation:Minimum=0
```

Comment:
If you have 0 as the minimum, and a default of 1, the user can't actually set 0 as the value unless you were to make this a pointer, so that there can be an unset nil value to be defaulted

Reply:
0 doesn't really makes sense anyway, we can raise the minimum to 1